File: ios/AlbBizMap_IOS/AlbBizMap/UI/Components/AuthGatePrompt.swift
// Bismillah Hir Rahman Nir Raheem
import SwiftUI

public struct AuthGatePrompt: View {
    @Environment(\.appStrings) private var strings
    public let isPresented: Bool
    public let onConfirm: () -> Void
    public let onDismiss: () -> Void
    
    public init(isPresented: Bool, onConfirm: @escaping () -> Void, onDismiss: @escaping () -> Void) {
        self.isPresented = isPresented
        self.onConfirm = onConfirm
        self.onDismiss = onDismiss
    }
    
    public var body: some View {
        if isPresented {
            ZStack {
                Color.black.opacity(0.4)
                    .edgesIgnoringSafeArea(.all)
                    .onTapGesture { onDismiss() }
                
                VStack(spacing: 16) {
                    Image(systemName: "lock.shield.fill")
                        .font(.system(size: 44))
                        .foregroundColor(.meTontRed)
                    
                    Text(strings.signIn)
                        .font(.title3)
                        .fontWeight(.bold)
                    
                    Text(strings.signInToContinue)
                        .font(.subheadline)
                        .foregroundColor(.gray)
                        .multilineTextAlignment(.center)
                    
                    HStack(spacing: 12) {
                        Button(action: onDismiss) {
                            Text(strings.notNow)
                                .fontWeight(.semibold)
                                .frame(maxWidth: .infinity)
                                .padding(.vertical, 12)
                                .background(Color.gray.opacity(0.15))
                                .foregroundColor(.meTontBlack)
                                .cornerRadius(10)
                        }
                        
                        Button(action: onConfirm) {
                            Text(strings.signIn)
                                .fontWeight(.semibold)
                                .frame(maxWidth: .infinity)
                                .padding(.vertical, 12)
                                .background(Color.meTontRed)
                                .foregroundColor(.white)
                                .cornerRadius(10)
                        }
                    }
                    .padding(.top, 8)
                }
                .padding(24)
                .background(Color.white)
                .cornerRadius(20)
                .shadow(radius: 10)
                .padding(.horizontal, 40)
            }
        }
    }
}

File: android/app/src/main/java/com/albbiz/map/ui/AuthGatePrompt.kt
// Bismillah Hir Rahman Nir Raheem
package com.albbiz.map.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp

// Contextual "sign in to continue" prompt shown in place of an instant
// redirect to the auth screen. Mirrors the iOS AuthGatePrompt design (dark
// scrim, centered white card, lock icon, Not Now / Sign In buttons) instead
// of the plain Material AlertDialog this used to be on Android — same
// signIn/signInToContinue/notNow strings, same confirm/dismiss behavior,
// just a UI pass so both platforms look consistent.
@Composable
fun AuthGatePrompt(
    onConfirm: () -> Unit,
    onDismiss: () -> Unit
) {
    val strings = LocalAppStrings.current

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(Color.Black.copy(alpha = 0.4f))
            .clickable(
                indication = null,
                interactionSource = remember { MutableInteractionSource() }
            ) { onDismiss() },
        contentAlignment = Alignment.Center
    ) {
        Card(
            modifier = Modifier
                .padding(horizontal = 40.dp)
                // Absorbs taps so they don't fall through to the scrim's
                // onDismiss above when tapping inside the card itself.
                .clickable(
                    indication = null,
                    interactionSource = remember { MutableInteractionSource() }
                ) {},
            shape = RoundedCornerShape(20.dp),
            colors = CardDefaults.cardColors(containerColor = Color.White),
            elevation = CardDefaults.cardElevation(defaultElevation = 10.dp)
        ) {
            Column(
                modifier = Modifier.padding(24.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                Icon(
                    imageVector = Icons.Default.Lock,
                    contentDescription = null,
                    tint = MeTontRed,
                    modifier = Modifier.size(44.dp)
                )

                Text(
                    strings.signIn,
                    style = MaterialTheme.typography.titleLarge,
                    fontWeight = FontWeight.Bold,
                    color = Color.Black
                )

                Text(
                    strings.signInToContinue,
                    style = MaterialTheme.typography.bodyMedium,
                    color = MeTontGrey,
                    textAlign = TextAlign.Center
                )

                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(top = 8.dp),
                    horizontalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    Button(
                        onClick = onDismiss,
                        modifier = Modifier
                            .weight(1f)
                            .height(46.dp),
                        shape = RoundedCornerShape(10.dp),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = Color.Black.copy(alpha = 0.08f),
                            contentColor = Color.Black
                        )
                    ) {
                        Text(strings.notNow, fontWeight = FontWeight.SemiBold)
                    }

                    Button(
                        onClick = onConfirm,
                        modifier = Modifier
                            .weight(1f)
                            .height(46.dp),
                        shape = RoundedCornerShape(10.dp),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = MeTontRed,
                            contentColor = Color.White
                        )
                    ) {
                        Text(strings.signIn, fontWeight = FontWeight.SemiBold)
                    }
                }
            }
        }
    }
}
